src/main/java/org/ibp/api/java/impl/middleware/common/validator/GermplasmUpdateValidator.java

```diff
@@ -0,0 +1,152 @@
+			this.userDefinedFieldService
+				.getByTableAndCodesInMap(UDTableType.ATRIBUTS_ATTRIBUTE.getTable(), new ArrayList<>(attributesAndNamesCodes));
+		final Map<String, Integer> nameCodes =
+			this.userDefinedFieldService
+				.getByTableAndCodesInMap(UDTableType.NAMES_NAME.getTable(), new ArrayList<>(attributesAndNamesCodes));
+
+		attributesAndNamesCodes.removeAll(attributeCodes.keySet());
+		attributesAndNamesCodes.removeAll(nameCodes.keySet());
+
+		if (!attributesAndNamesCodes.isEmpty()) {
+			errors.reject("germplasm.update.invalid.attribute.or.name.code", new String[] {String.join(",", attributesAndNamesCodes)}, "");
+		}
+
+		final Collection<String> ambiguosCodes = CollectionUtils.intersection(attributeCodes.keySet(), nameCodes.keySet());
+		if (!ambiguosCodes.isEmpty()) {
+			errors.reject("germplasm.update.ambiguous.code", new String[] {String.join(",", ambiguosCodes)}, "");
+		}
+
+	}
+
+	public void validateGermplasmIdAndGermplasmUUID(final BindingResult errors, final List<GermplasmUpdateDTO> germplasmUpdateDTOList) {
+		// Find rows (GermplasmUpdateDTO) with blank GID and UUID.
+		final Optional<GermplasmUpdateDTO> optionalGermplasmUpdateDTO =
+			germplasmUpdateDTOList.stream().filter(dto -> StringUtils.isEmpty(dto.getGermplasmUUID()) && dto.getGid() == null).findAny();
+		if (optionalGermplasmUpdateDTO.isPresent()) {
+			errors.reject("germplasm.update.missing.gid.and.uuid", "");
+		}
+
+		final Set<Integer> gids = germplasmUpdateDTOList.stream().map(dto -> dto.getGid()).collect(Collectors.toSet());
+		final Set<String> germplasmUUIDs = germplasmUpdateDTOList.stream().map(dto -> dto.getGermplasmUUID()).collect(Collectors.toSet());
+
+		final List<Germplasm> germplasmByGIDs = this.germplasmDataManager.getGermplasms(new ArrayList<>(gids));
+		final List<Germplasm> germplasmByUUIDs = this.germplasmDataManager.getGermplasmByUUIDs(germplasmUUIDs);
+
+		gids.removeAll(germplasmByGIDs.stream().map(dto -> dto.getGid()).collect(Collectors.toSet()));
+		germplasmUUIDs.removeAll(germplasmByUUIDs.stream().map(dto -> dto.getGermplasmUUID()).collect(Collectors.toSet()));
+
+		if (!gids.isEmpty()) {
+			errors.reject("germplasm.update.invalid.gid", new String[] {
+				String.join(",", gids.stream().map(o -> String.valueOf(o)).collect(
+					Collectors.toSet()))}, "");
+		}
+
+		if (!germplasmUUIDs.isEmpty()) {
+			errors.reject("germplasm.update.invalid.uuid", new String[] {String.join(",", germplasmUUIDs)}, "");
+		}
+
+	}
+
+	public void validateLocationAbbreviation(final BindingResult errors, final List<GermplasmUpdateDTO> germplasmUpdateDTOList) {
+
+		final Set<String> locationAbbreviationSet =
+			germplasmUpdateDTOList.stream().filter(dto -> StringUtils.isNotEmpty(dto.getLocationAbbreviation()))
+				.map(dto -> dto.getLocationAbbreviation()).collect(Collectors.toSet());
+		final List<String> abbreviations =
+			this.locationDataManager.getLocationsByAbbreviation(locationAbbreviationSet).stream().map(loc -> loc.getLabbr()).collect(
```

Comment:
Please consider that we need to use the programUUID to validate the locations, because the list changes depending the program and user that does not have access to an specific program location should not be using that code. The same with breeding methods. @abatac

Reply:
Noted on this. I'll update this once your branch is merged.

Reply:
Done.